packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -194,328 +112,73 @@ class FlutterDriver {
-  rpc.Peer _peer;
-
-  String _dartVmReconnectUrl;
-
-  Future<void> _restorePeerConnectionIfNeeded() async {
-    if (!_peer.isClosed || _dartVmReconnectUrl == null) {
-      return;
-    }
-
-    _log.warning(
-        'Peer connection is closed! Trying to restore the connection...'
+    return VMServiceFlutterDriver.connect(
+              dartVmServiceUrl: dartVmServiceUrl,
+              printCommunication: printCommunication,
+              logCommunicationToFile: logCommunicationToFile,
+              isolateNumber: isolateNumber,
+              fuchsiaModuleTarget: fuchsiaModuleTarget,
     );
-
-    final String webSocketUrl = _getWebSocketUrl(_dartVmReconnectUrl);
-    final WebSocket ws = await WebSocket.connect(webSocketUrl);
-    ws.done.whenComplete(() => _checkCloseCode(ws));
-    _peer = rpc.Peer(
-        IOWebSocketChannel(ws).cast(),
-        onUnhandledError: _unhandledJsonRpcError,
-    )..listen();
   }
 
-  /// The main isolate hosting the Flutter application.
-  ///
-  /// If you used the [registerExtension] API to instrument your application,
-  /// you can use this [VMIsolate] to call these extension methods via
-  /// [invokeExtension].
-  final VMIsolate appIsolate;
-
-  /// Whether to print communication between host and app to `stdout`.
-  final bool _printCommunication;
-
-  /// Whether to log communication between host and app to `flutter_driver_commands.log`.
-  final bool _logCommunicationToFile;
-
-  Future<Map<String, dynamic>> _sendCommand(Command command) async {
-    Map<String, dynamic> response;
-    try {
-      final Map<String, String> serialized = command.serialize();
-      _logCommunication('>>> $serialized');
-      final Future<Map<String, dynamic>> future = appIsolate.invokeExtension(
-        _flutterExtensionMethodName,
-        serialized,
-      ).then<Map<String, dynamic>>((Object value) => value);
-      response = await _warnIfSlow<Map<String, dynamic>>(
-        future: future,
-        timeout: command.timeout ?? kUnusuallyLongTimeout,
-        message: '${command.kind} message is taking a long time to complete...',
-      );
-      _logCommunication('<<< $response');
-    } catch (error, stackTrace) {
-      throw DriverError(
-        'Failed to fulfill ${command.runtimeType} due to remote error',
-        error,
-        stackTrace,
-      );
-    }
-    if (response['isError'])
-      throw DriverError('Error in Flutter application: ${response['response']}');
-    return response['response'];
-  }
-
-  void _logCommunication(String message) {
-    if (_printCommunication)
-      _log.info(message);
-    if (_logCommunicationToFile) {
-      final f.File file = fs.file(p.join(testOutputsDirectory, 'flutter_driver_commands_$_driverId.log'));
-      file.createSync(recursive: true); // no-op if file exists
-      file.writeAsStringSync('${DateTime.now()} $message\n', mode: f.FileMode.append, flush: true);
-    }
-  }
+  /// Sends [command] to the Flutter Driver extensions.
```

Comment:
Though this wasn't extensively documented before, you might add some additional comments that this must be implemented by subclasses, and give some breadcrumbs to the concrete implementations as examples.